ssl/quic/quic_port.c

```diff
@@ -551,6 +552,7 @@ static QUIC_CHANNEL *port_make_channel(QUIC_PORT *port, SSL *tls, OSSL_QRX *qrx,
      * And finally init the channel struct
      */
     if (!ossl_quic_channel_init(ch)) {
+        OPENSSL_free(ch->qlog_title);
         SSL_free(ch->tls);
```

Comment:
I understand line 556 is not your change, it's present in existing code. I think it is actually bug to do SSL_free() here. The reasons why:
  - the `ch_cleanup()` function we call in regular code paths does not call `SSL_free(ch->tls);`, so we should not be calling `OPENSSL_free()` here as well.
  - I suspect the `ch->tls` points to tls object _owned_ by SSL which also owns the channel.